test/fx/test_subgraph_rewriter.py

```diff
@@ -0,0 +1,188 @@
+            t1 = torch.sum(w1, 1)
+            t2 = torch.addmm(b1, m1, m2.t())
+            return torch.eq(torch.sum(t1), torch.sum(t2))
+
+        # Pattern we want to match in `fn`
+        def pattern(x, w1, w2, b1, b2):
+            m1 = torch.cat([w1, w2])
+            m2 = torch.cat([x, b2])
+            return torch.addmm(b1, m1, m2.t())
+
+        # Set up
+        traced_fn = symbolic_trace(fn)
+        comparison_fn = symbolic_trace(fn)
+
+        # Auxiliary testing variables
+        x = torch.rand(1, 3)
+        w1 = torch.rand(1, 3)
+        w2 = torch.rand(1, 3)
+        b1 = torch.rand(2, 2)
+        b2 = torch.rand(1, 3)
+
+        # Replace `pattern` with the same pattern
+        subgraph_rewriter.replace_pattern(traced_fn.graph, pattern, pattern)
+
+        # Check that the graph is still valid after the rewrite
+        traced_fn.graph.lint(traced_fn)
+
+        # Ensure that the graph is functionally the same as it was
+        # before the rewrite
+        ref_outs = comparison_fn(x, w1, w2, b1, b2)
+        test_outs = traced_fn(x, w1, w2, b1, b2)
+        self.assertEqual(ref_outs, test_outs)
+
+    def test_subgraph_rewriter_with_oneliner_pattern(self):
+        # Function that we want to perform the rewrite on
+        def fn(x, w1, w2, b1, b2):
+            m1 = torch.cat([w1, w2])
+            m2 = torch.cat([x, b2])
+            t1 = torch.sum(w1, 1)
+            t2 = torch.addmm(b1, m1, m2.t())
+            return torch.eq(torch.sum(t1), torch.sum(t2))
+
+        # Pattern we want to match in `fn`
+        def pattern(x, w1, w2, b1, b2):
+            return torch.cat([x, b2])
+
+        # What we want to replace `pattern` with
+        def replacement(x, w1, w2, b1, b2):
+            return torch.cat([b2, w2])
+
+        # Set up
+        traced_fn = symbolic_trace(fn)
+        subgraph_rewriter.replace_pattern(traced_fn.graph, pattern, replacement)
+
+        # Check that the graph is still valid after the rewrite
+        traced_fn.graph.lint(traced_fn)
+
+    def test_subgraph_rewriter_single_pattern_match(self):
+        # Function that we want to perform the rewrite on
+        def fn(x, w1, w2, b1, b2):
+            m1 = torch.cat([w1, w2])
+            m2 = torch.cat([x, b2])
+            t1 = torch.sum(w1, 1)
+            t2 = torch.addmm(b1, m1, m2.t())
+            return torch.eq(torch.sum(t1), torch.sum(t2))
+
+        # Pattern we want to match in `fn`
+        def pattern(x, w1, w2, b1, b2):
+            m1 = torch.cat([w1, w2])
+            m2 = torch.cat([x, b2])
+            return torch.addmm(b1, m1, m2.t())
+
+        # What we want to replace `pattern` with
+        def replacement(x, w1, w2, b1, b2):
+            m2 = torch.cat([x, b2])
+            m1 = torch.cat([w1, w2])
+            more_lines = torch.mm(w1, w2.t())
+            return torch.addmm(b1, m1, m2.t())
+
+        # Set up
+        traced_fn = symbolic_trace(fn)
+        comparison_fn = symbolic_trace(fn)
```

Comment:
`comparison_fn` is equivalent to the original `traced_fn`. Because of the `recompile()` issue above, both `comparison_fn` and `traced_fn` are equivalent, but that's actually not correct, we want to test that they both compute the updated code.

I think you'll need to define a separate reference function, e.g.:

```
def reference(x, w1, w2, b1, b2):
  t1 = torch.sum(w1, 1)
  m2 = torch.cat([x, b2])
  m1 = torch.cat([w1, w2])
  more_lines = torch.mm(w1, w2.t())
  t2 = torch.addmm(b1, m1, m2.t())
  return torch.eq(torch.sum(t1), torch.sum(t2))
```
(or whatever it's supposed to be)

and then `comparison_function = symbolic_trace(reference)`